gulpfile.js

```diff
@@ -0,0 +1,381 @@
+'use strict';
+/**
+ * This file needs to be JavaScript and is read by gulp.
+ */
+// Global imports.
+const child_process = require('child_process');
+const fs = require('fs');
+const gulp = require('gulp');
+const path = require('path');
+const resolveBin = require('resolve-bin');
+
+// Other imports.
+const inlineResources = require('./scripts/release/inline-resources');
+const karma = require('karma');
+
+// Gulp plugins.
+const gulpClean = require('gulp-clean');
+const gulpServer = require('gulp-server-livereload');
+const gulpMerge = require('merge2');
+const gulpRunSequence = require('run-sequence');
+const gulpSass = require('gulp-sass');
+const gulpSourcemaps = require('gulp-sourcemaps');
+const gulpTs = require('gulp-typescript');
+
+
+// Directories.
+const srcDir = path.join(__dirname, 'src');
+const componentsDir = path.join(srcDir, 'lib');
+const devAppDir = path.join(srcDir, 'demo-app');
+const e2eAppDir = path.join(srcDir, 'e2e-app');
+
+const outDir = 'dist';
+const outLibDir = path.join(outDir, '@angular2-material');
+
+
+/**
+ * Create a TS Build Task, based on the options.
+ */
+function createTsBuildTask(options) {
+  const tsConfigDir = options.tsConfigPath;
+  const tsConfigPath = path.join(tsConfigDir, 'tsconfig.json');
+  const tsConfig = JSON.parse(fs.readFileSync(tsConfigPath, 'utf-8'));
+  const dest = path.join(tsConfigDir, tsConfig.compilerOptions.outDir);
+
+  return function() {
+    const tsProject = gulpTs.createProject(tsConfigPath, {
+      typescript: require('typescript')
+    });
+
+    let pipe = tsProject.src()
+      .pipe(gulpSourcemaps.init())
```

Comment:
It's technically not `tsc`. It's its own gulp typescript plugin, and it removes the sourcemap from the tsconfig before creating its program.